### DISCUSSION TITLE
JK BMS CanBUS Protocol

### DISCUSSION THREAD

**Discussion by karikolehmainen** [Louisvdw/dbus-serialbattery#1123] [General]
Hello,

I have a quick question about JK-BMS CanBUS protocol. I have BMS connected to Victron Cerbo via BMS.Can and I see that packets are received on Cerbo. It obviously can’t understand the protocol because I do not see the battery in devices. Does this software also support BMS.Bus for communicating with JK BMS?

**Comment by mr-manuel** [comment]
https://mr-manuel.github.io/venus-os_dbus-serialbattery_docs/general/features#bms-feature-comparison

**Comment by karikolehmainen** [comment]
Thanks. Seems that there is some problem with the CAN bus communication though. I have JK-BMS B2A20S20PR that has CAN bus enabled. With mobile App you can choose from two protocol options one being 250kbit/s and other 500kbit/s and both V2 of JK BMS Can protocol. I can see from the Cerbo that there are stuff moving in the VE.Can when set 250 setting both and also in BMS.Can if set to 500kbit/s. dbus-serialbattery is not able to detect any batteries:
````
root@einstein:~# tail -f /data/log/dbus-canbattery.can1/current 
@4000000066db2f8404fe05d4 INFO:SerialBattery:dbus-serialbattery v1.3.20240705
@4000000066db2f960e43737c INFO:SerialBattery:-- Testing BMS: 1 of 3 rounds
@4000000066db2f962c356fac INFO:SerialBattery:-- Testing BMS: 2 of 3 rounds
@4000000066db2f970e7d7edc INFO:SerialBattery:-- Testing BMS: 3 of 3 rounds
@4000000066db2f972c64018c ERROR:SerialBattery:ERROR >>> No battery connection at can1
@4000000066db2f9802dc7d94 
@4000000066db2f99209fb904 INFO:SerialBattery:
@4000000066db2f9920aa30b4 INFO:SerialBattery:Starting dbus-serialbattery
@4000000066db2f9920cbc65c INFO:SerialBattery:Venus OS v3.41
@4000000066db2f9920d5ad84 INFO:SerialBattery:dbus-serialbattery v1.3.20240705
^C
root@einstein:~# tail -f /data/log/dbus-canbattery.can0/current 
@4000000066db2f191191bfd4 INFO:SerialBattery:-- Testing BMS: 1 of 3 rounds
@4000000066db2f192f786d7c INFO:SerialBattery:-- Testing BMS: 2 of 3 rounds
@4000000066db2f1a11c27c64 INFO:SerialBattery:-- Testing BMS: 3 of 3 rounds
@4000000066db2f1a2fa53654 ERROR:SerialBattery:ERROR >>> No battery connection at can0
@4000000066db2f1b078748c4 
@4000000066db2f1c230d8c84 INFO:SerialBattery:
@4000000066db2f1c231b9644 INFO:SerialBattery:Starting dbus-serialbattery
@4000000066db2f1c233b5b14 INFO:SerialBattery:Venus OS v3.41
@4000000066db2f1c2348574c INFO:SerialBattery:dbus-serialbattery v1.3.20240705
@4000000066db2f283693d31c 
^C
````
With bluetooth the module works OK though. It can be that the cable that I made is not OK, but I would think that it is fine as I can see whole bunch of packets received in Cerbo UI... Could it be that this V2 of the JK Can protocol is not compatible?

**Reply by mr-manuel** [reply]
Please open an issue in this repository: https://github.com/mr-manuel/venus-os_dbus-serialbattery/issues/new/choose

Please read carefully the instructions and don't skip anything.

**Comment by haraldhh** [comment]
I found this which is the same issue I have basically, but no following issue so I opened one.

https://github.com/mr-manuel/venus-os_dbus-serialbattery/issues/149
